apps/allocations/app/components/App/AllocationsHistory.js

```diff
@@ -63,30 +62,34 @@ const AllocationsHistory = ({ allocations }) => {
           </Amount>
         ]
       }}
-      //renderEntryExpansion={({ recipients, amount, token }) => {
-      //  return recipients.map((recipient, index) => {
-      //    const allocated = BigNumber(recipient.amount).div(amount)
-      //    return (
-      //      <Recipient key={index}>
-      //        <IdentityBadge
-      //          entity={recipient.address}
-      //          shorten={true}
-      //        />
-      //        <RecipientProgress>
-      //          <ProgressBar
-      //            value={allocated.toNumber()}
-      //            color={String(theme.accentEnd)}
-      //          />
-      //        </RecipientProgress>
-      //        <RecipientAmount theme={theme}>
-      //          { displayCurrency(BigNumber(recipient.amount)) } {' '}
-      //          {token} {' • '}
-      //          { allocated.times(100).dp(0).toNumber() }{'%'}
-      //        </RecipientAmount>
-      //      </Recipient>
-      //    )
-      //  })
-      //}}
+      renderEntryExpansion={({ recipients, amount, token }) => {
+        const totalSupports = recipients.reduce((total, recipient) => {
+          return total + Number(recipient.supports)
+        }, 0)
+        return recipients.map((recipient, index) => {
+          const allocated = BigNumber(recipient.supports).div(totalSupports)
```

Comment:
In this case, let's keep this here, but for the future, we might want to avoid adding any kind of logic inside the return of the render function, and calculate any value outside of that instead, to prevent unnecessary extra re-renders and also to make our render functions more performant.

Reply:
@ottodevs I agree with that general sentiment, but I'm not sure it's useful or even possible here. Where do you picture moving this calculation?

Reply:
The idea is to move outside the return, so any change to it does not force a re-render, and is possible to event memoize the values in future optimizations, but as said, is just a general suggestion, no need to change it here for now, thanks!